keps/sig-api-machinery/0006-apply.md

```diff
@@ -194,8 +195,27 @@ atomic. That can be specified with `// +mapType=atomic` or `//
 +structType=atomic` respectively. They map to the same openapi extension:
 `"x-kubernetes-map-type": "atomic"`.
 
-#### Kubectl
-##### Server-side Apply
+#### Clients
+
+##### Apply Status Header
+
+Client side apply in kubectl provides the user with information if the applied config resulted in any changes
+on the apiserver.
+
+To provided the same information with server-side apply, a dedicated header is set by the apiserver
+if the request caused changes to the existing resource.
+
+```txt
+X-Kubernetes-Apply-Status: "modified" || "unmodified"
+```
+
+This header can then be used by e.g. `kubectl` to provide the user with the desired information (rel: [#85750](https://github.com/kubernetes/kubernetes/issues/85750), [#66450](https://github.com/kubernetes/kubernetes/issues/66450)).
+
+An alternative that was considered is a http status code, indicating the resource was not modified.
+This approach was discarded due to the lack of a matching 2xx status code.
```

Comment:
Please add a paragraph to the effect that the intended use of this is informational only, and it's buggy to e.g. avoid doing the next step in a multi-step deployment based on the value of this header.